### PR TITLE
Fixes #20857 - override post_sync_url

### PR DESCRIPTION
**Pull request by sean797**

### CODE REVIEW

**Comment by ekohl** [comment]
What do you think about using ```$::foreman::foreman_url```? An alternative is re-using ```$::foreman::servername```. It's messy we have both and I'd prefer ```$::foreman::foreman_url```. The benefit is that when you'd include ```foreman::cli``` on the same host you automatically configure that correct (though I don't think katello-installer exposes that option).

**Comment by sean797** [comment]
@ekohl Yea, nice idea. Updated to use `$::foreman::foreman_url` Thanks!

File: manifests/application.pp [hunk 1]
```diff
@@ -17,11 +17,13 @@
   Optional[String] $proxy_username = $::katello::proxy_username,
   Optional[String] $proxy_password = $::katello::proxy_password,
 ) {
+  include ::foreman
   include ::certs
   include ::certs::apache
   include ::certs::foreman
   include ::certs::pulp_client
 
+  $application_host = $::foreman::foreman_url
```
Comment (ekohl) [inline manifests/application.pp]: It might simplify the config template if you define ```$post_sync_url``` here.

**Comment by sean797** [comment]
Hows that @ekohl ?

File: manifests/application.pp [hunk 2]
```diff
@@ -17,11 +17,14 @@
...
   Optional[String] $proxy_password = $::katello::proxy_password,
 ) {
+  include ::foreman
   include ::certs
   include ::certs::apache
   include ::certs::foreman
   include ::certs::pulp_client
 
+  $post_sync_url = "${$::foreman::foreman_url}${deployment_url}/api/v2/repositories/sync_complete?token=${post_sync_token}"
+  $application_host = $::foreman::foreman_url
```
Comment (ekohl) [inline manifests/application.pp]: Is this used anywhere?

**Comment by ekohl** [comment]
This looks a lot like a change I already wanted to make but hadn't gotten around to :)

**Comment by sean797** [comment]
ah, damn! That was left over from before. Thanks @ekohl

File: manifests/application.pp [hunk 3]
```diff
@@ -17,11 +17,13 @@
   Optional[String] $proxy_username = $::katello::proxy_username,
   Optional[String] $proxy_password = $::katello::proxy_password,
 ) {
+  include ::foreman
   include ::certs
   include ::certs::apache
   include ::certs::foreman
   include ::certs::pulp_client
 
+  $post_sync_url = "${$::foreman::foreman_url}${deployment_url}/api/v2/repositories/sync_complete?token=${post_sync_token}"
```
Comment (ekohl) [inline manifests/application.pp]: I'm surprised this works. Usually we have ```${::foreman::foreman_url}```. The tests appear to pass so this is valid syntax?

Comment (sean797) [reply manifests/application.pp]: Interesting, it must be. I was actually a typo so I've update to match the rest of our code base.

**Comment by ekohl** [comment]
Thanks! This removes an item from my (mental) list of things to fix.

**Pull request merged**